### PR TITLE
feat(tx-payment): add wrapper boiler 

### LINKED ISSUE #1024: [Capacity] [SubTask-1-of-4] new wrapper pallet, `frequency-transaction-payment` for transactions payments which is just a passthrough
This is a subtask of #1010 as the parent see for details

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Add boilerplate for submitting transactions
with both Capacity and Token accounts.
This boilerplate pallet includes a SignedExtension
that wraps the Substrate transaction-payment pallet.

Closes #1024

### CODE REVIEW

File: js/api-augment/index.ts [hunk 1]
```diff
@@ -8,20 +8,23 @@ import * as definitions from "./interfaces/definitions";
 /**
  * Build up the types for ApiPromise.create
  */
-export const types = Object.entries(definitions).reduce((acc, [_key, value]) => {
-  return {
-    ...acc,
-    ...value.types,
-  };
-}, {});
+export const types = Object.entries(definitions).reduce(
```
Comment (enddynayn) [inline js/api-augment/index.ts:11]: we can remove this

File: js/api-augment/index.ts [hunk 2]
```diff
@@ -32,19 +35,28 @@ export const signedExtensions: ExtDef = {
   // `CheckFreeExtrinsicUse` has no payload or extrinsic requirements
   CheckFreeExtrinsicUse: {
     extrinsic: {},
-    payload: {},
+    payload: {}
   },
+  ChargeFrqTxPayment: {
```
Comment (enddynayn) [inline js/api-augment/index.ts:40]: we can remove this

**Comment by enddynayn** [comment]
Since building it is taking forever, I am now opening it up for review.
Pending todo:
1. better commit message
2. squash commits
3. remove api-augment code I added.

Please ask questions.

File: pallets/frequency-tx-payment/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+[package.metadata.docs.rs]
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+codec = {package = "parity-scale-codec", version = "3.1.5", default-features = false, features = [
+  "derive",
+]}
+common-primitives = {default-features = false, path = "../../common/primitives"}
+frame-benchmarking = {default-features = false, git = "https://github.com/paritytech/substrate.git", optional = true, branch = "polkadot-v0.9.36"}
```
Comment (johnmela) [inline pallets/frequency-tx-payment/Cargo.toml:20]: We can remove `.git` these substrate urls

**Review by johnmela — APPROVED** [review]

File: pallets/frequency-tx-payment/src/tests.rs [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	ExtBuilder::default()
+		.balance_factor(balance_factor)
+		.base_weight(Weight::from_ref_time(5))
+		.build()
+		.execute_with(|| {
+			let account_id = 1u64;
+			let balances_call: &<Test as frame_system::Config>::RuntimeCall =
+				&RuntimeCall::Balances(BalancesCall::transfer { dest: 2, value: 100 });
+			let dispatch_info =
+				DispatchInfo { weight: Weight::from_ref_time(5), ..Default::default() };
```
Comment (shannonwells) [inline pallets/frequency-tx-payment/src/tests.rs:116]: 🏆

**Review by shannonwells — APPROVED** [review]
Slick!

**Pull request merged**